extensions/sources/base/src/main/java/io/smallrye/ext/config/source/base/BaseConfigSource.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.smallrye.ext.config.source.base;
+
+import java.util.logging.Level;
+import lombok.Getter;
+import lombok.extern.java.Log;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+/**
+ * Base class for all our ConfigSources (Originally from Geronimo Config)
+ *
+ * @author <a href="mailto:struberg@apache.org">Mark Struberg</a>
+ * @author <a href="mailto:gpetracek@apache.org">Gerhard Petracek</a>
+ * @author <a href="mailto:dpmoore@acm.org">Derek P. Moore</a>
+ */
+@Log
+public abstract class BaseConfigSource implements ConfigSource {
+    
+    @Getter
+    private final Config config;
+    private int ordinal = 1000; // default
```

Comment:
I'm not sure it makes sense to allow the ordinal to be mutable?

Reply:
Can you please explain why you say that ? This allows a user to change the ordinal with a config value.